submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,338 @@
+				</div>
+				<div class="card__row country__row">
+					<h6 class="card__country">${friend.country}</h6>
+				</div>
+			</div>`;
+}
+
+function reformatPhoneNumber(number) {
+	return number
+		.replace(/[^0-9]+/g, "")
+		.replace(/.(\d{3})/g, "$1-")
+		.replace(/(^\d{3,3})-(.\d+)/, "+($1)-$2")
+		.replace(/[-]+$/g, "");
+}
+
+function getDate(date) {
+	return new Date(date).toLocaleString("en-US", {
+		year: "numeric",
+		month: "long",
+		day: "numeric",
+	});
+}
+
+function getGenderIcon(gender) {
+	return gender === "female"
+		? '<span class="female">♀</span>'
+		: '<span class="male">♂</span>';
+}
+
+function getResponseErrorMessage(status, statusText) {
+	return `<h1 class='error__message'>Sorry, an error occured!</h1>
+			<h2 class='error__code'>${status}: ${statusText}</h2>`;
+}
+
+function setAgeLimit(ageLimit, limitInputId, limitType) {
+	const limitInput = document.getElementById(limitInputId);
+	limitInput[limitType] = ageLimit;
+	setInputLimitValue(
+		limitInput,
+		limitInputId,
+		limitType,
+		"minAge",
+		"min",
+		ageLimit
+	);
+	setInputLimitValue(
+		limitInput,
+		limitInputId,
+		limitType,
+		"maxAge",
+		"max",
+		ageLimit
+	);
+}
+
+function setInputLimitValue(
+	limitInput,
+	limitInputId,
+	limitType,
+	limitInputIdValue,
+	limitTypeValue,
+	ageLimit
+) {
+	if (limitInputId === limitInputIdValue && limitType === limitTypeValue)
+		limitInput.value = ageLimit;
+}
+
+function getCertainAgeLimit(friendsArray, value) {
+	const sortedArray = friendsArray.sort((a, b) => a.age - b.age);
+	return value === "min"
+		? sortedArray[0].age
+		: sortedArray[sortedArray.length - 1].age;
+}
+
+function sortCards(e, friendsArray) {
+	if (e.target.classList.contains("list__item")) {
+		document.querySelector(".select__face-item").innerText =
+			e.target.textContent;
+		sortCardsArray(e.target.attributes.value.value, friendsArray);
+	}
+}
+
+function sortCardsArray(condition, friendsArray) {
+	switch (condition) {
+		case "ND":
+			FRIENDS_ARRAY.sort((a, b) => b.firstName.localeCompare(a.firstName));
+			break;
+		case "NA":
+			FRIENDS_ARRAY.sort((a, b) => a.firstName.localeCompare(b.firstName));
+			break;
+		case "AD":
+			FRIENDS_ARRAY.sort((a, b) => b.age - a.age);
+			break;
+		case "AA":
+			FRIENDS_ARRAY.sort((a, b) => a.age - b.age);
+			break;
+	}
+}
+
+function findSubstring(string, substring) {
+	return string.toLowerCase().indexOf(substring.toLowerCase()) >= 0
+		? true
+		: false;
+}
+
+function findMatchesWithPropertiesValues(
+	propertiesList,
+	friendsArray,
+	substring
+) {
+	return friendsArray.filter((friend) => {
+		return propertiesList
+			.map((property) => findSubstring(friend[property], substring))
+			.some((el) => el);
+	});
+}
+
+function filterByGenderValue(genderList, friendsArray) {
+	return friendsArray.filter((friend) =>
+		genderList.some((gender) => friend.gender === gender.value)
+	);
+}
+
+function filterByAgeLimits(minAge, maxAge, friendsArray) {
+	return friendsArray.filter(
```

Comment:
Move it into `filterByAge`. Occam blade - remove redundant entities.

Reply:
I've moved this one inside the `filterByAge`